os/arch/arm/src/amebasmart/amebasmart_spi.c

```diff
@@ -857,8 +858,10 @@ static uint16_t amebasmart_spi_send(FAR struct spi_dev_s *dev, uint16_t wd)
 	DEBUGASSERT(priv);
 
 	if (priv->role == AMEBASMART_SPI_MASTER) {
-
-		ret = spi_master_write(&priv->spi_object, wd);
+		if (priv->mtr_wrrd_disable)
+			ret = spi_master_write(&priv->spi_object, wd, 1);
+		else
+			ret = spi_master_write(&priv->spi_object, wd, 0);
```

Comment:
I think this will affect other devices that are connected to the same spi port that use both tx and rx. So, this will not be a good approach. @Taejun-Kwon could you also please take a look at this change?

Reply:
@abhishek-samsung @kaikaikaime @Taejun-Kwon
What I think is
1. The write function should do only write operation without read.
2. The exchange function should do both write and read.

So, LCD driver will use the write, not the exchange because the LCD does not support the read.
@abhishek-samsung @Taejun-Kwon Do you need the read at the write function? Can't you use the exchange?

@kaikaikaime If @abhishek-samsung and @Taejun-Kwon can use the exchange instead of the write, then, the write should do write only regardless of the config. (No need to add the config)

Please let me know your opinion.

Reply:
We should concern that some SPI device does not support read operation.

Reply:
```
static const struct spi_ops_s g_spi1ops = {
...
	.send         = amebalite_spi_send,
#ifdef CONFIG_SPI_EXCHANGE
	.exchange     = amebalite_spi_exchange,
#else
	.sndblock     = amebalite_spi_sndblock,
	.recvblock    = amebalite_spi_recvblock,
#endif
...
}
```
There is send api without exchange in SPI ops interface

How about support send of amebasmart_spi which not related to exchange.

Reply:
Currently when spi exchange config is enabled, snd and rcv are both use exchange internally. So, it needs to be split accordingly and we need to only modify the snd api of SPI without changing anything else.
https://github.com/Samsung/TizenRT/blob/1601a36b77dc7bdee54f3b72876213b6b56c2e8e/os/include/tinyara/spi/spi.h#L281-L285
So, I think instead of a config, it would be good if @kaikaikaime would provide us a snd API and plug it in ```amebalite_spi_sndblock``` like Eunwoo suggested. Then we will need to make the upper layer changes on our side.

Reply:
@kaikaikaime Sorry but this is urgent. Could you make a change ASAP?

Reply:
Hi all, based on the earlier discussion, this will be my following actions, do correct me if I get them wrong:
1. I will create another internal write function which does not run read function ()
2. Using this new internal write and place them in SPI_SNDBLOCK when CONFIG_SPI_EXCHANGE is disabled.

Kai

Reply:
With a new creation of spi master send with read remove, I can now place under the condition when exchange is undefined. Change is minimal.